### PR TITLE
board: function renaming / defining for LEDs and Keys to new API.

### PR DESCRIPTION
**Pull request by RomanBuchert**
Described in #15

### CODE REVIEW

**Comment by tuupola** [comment]
https://doc.nucleisys.com/nuclei_sdk/contribute.html#git-commit-guide

Commit messages should be present tense in imperative mood. Ie commit message should answer to question "Applying this PR will: xxx".  For example "Applying this PR will: **Correct Indentation** " or  "Applying this PR will: **Rename functions according to new API** "

This is needed so that project will have a nice clean commit history.

File: SoC/gd32vf103/Board/gd32vf103v_rvstar/Include/gd32vf103v_rvstar.h [hunk 1]
```diff
@@ -113,6 +116,16 @@ void gd_rvstar_key_init(key_typedef_enum keynum, keymode_typedef_enum keymode);
 uint8_t gd_rvstar_key_state_get(key_typedef_enum keynum);
 /* configure COM port */
 void gd_com_init(uint32_t usart_periph);
+
+/* Defines for LED functio/* Defines for LED / Key functions to new / general API */
+#define gd_led_init         gd_rvstar_led_init
```
Comment (fanghuaqi) [inline SoC/gd32vf103/Board/gd32vf103v_rvstar/Include/gd32vf103v_rvstar.h:121]: I prefer to change the old style API to new ones, and use macros to be back compatiable with previous.

In future, we can slowly deprecate the old APIs.

File: SoC/gd32vf103/Board/gd32vf103v_eval/Include/gd32vf103v_eval.h [hunk 2]
```diff
@@ -161,6 +165,15 @@ uint8_t gd_eval_key_state_get(key_typedef_enum key);
 /* configure COM port */
 void gd_com_init(uint32_t com);
 
+/* Defines for LED / Key functions to new / general API */
+#define gd_led_init         gd_eval_led_init
```
Comment (fanghuaqi) [inline SoC/gd32vf103/Board/gd32vf103v_eval/Include/gd32vf103v_eval.h:169]: Same as rvstar

**Review by fanghuaqi — CHANGES_REQUESTED** [review]
Change old rvstar and eval board APIs to new APIs, and use macros to be back compatiable with old ones.

File: SoC/gd32vf103/Board/gd32vf103v_eval/Source/gd32vf103v_eval.c [hunk 3]
```diff
@@ -1,252 +1,252 @@
...
+    /* enable the key clock */
+    rcu_periph_clock_enable(KEY_CLK[keynum]);
+    rcu_periph_clock_enable(RCU_AF);
+
+    /* configure button pin as input */
+    gpio_init(KEY_PORT[keynum], GPIO_MODE_IN_FLOATING, GPIO_OSPEED_50MHZ, KEY_PIN[keynum]);
+
+    if (key_mode == KEY_MODE_EXTI) {
+        /* enable and set key EXTI interrupt to the lowest priority */
+	ECLIC_EnableIRQ(KEY_IRQn[keynum]);
```
Comment (fanghuaqi) [inline SoC/gd32vf103/Board/gd32vf103v_eval/Source/gd32vf103v_eval.c:183]: This code is not aligned, could you help to align it, using 4 spaces.

**Review by fanghuaqi — CHANGES_REQUESTED** [review]
Thanks for your update.

Could you also help to format the code of old gd32v board(rvstar and eval) in this PR, there are some code lines not aligned, and some code are mixed with space and tab, it is included previously by me, not a good start.

Thanks
Huaqi

File: SoC/gd32vf103/Board/gd32vf103v_eval/Source/gd32vf103v_eval.c (hunk 3 above)
Comment (RomanBuchert) [reply SoC/gd32vf103/Board/gd32vf103v_eval/Source/gd32vf103v_eval.c:183]: grrr. I don't know, why this always happens. I've extra running your editor config in notepad++.

**Comment by RomanBuchert** [comment]
> Thanks for your update.
> 
> Could you also help to format the code of old gd32v board(rvstar and eval) in this PR, there are some code lines not aligned, and some code are mixed with space and tab, it is included previously by me, not a good start.
> 
> Thanks
> Huaqi

... I hope I got all the TABs.

**Review by fanghuaqi — APPROVED** [review]

**Pull request merged**